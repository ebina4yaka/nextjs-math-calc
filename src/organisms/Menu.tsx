import React, { useEffect, useState } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import ContentCard from '../molecules/ContentCard'
import contentList, { Content } from '../data/contentList'

const useStyles = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexWrap: 'wrap',
  },
})

export default function Menu(): React.ReactElement {
  const [contents, setContents] = useState(Array<Content>())
  const classes = useStyles()

  useEffect(() => {
    setContents(contentList)
  }, [])

  const createContents = (): JSX.Element[] => {
    const list: JSX.Element[] = []
    contents.map((content: Content) =>
      list.push(
        <ContentCard
          key={content.id}
          title={content.title}
          url={content.url}
          image={content.image}
        />
      )
    )
    return list
  }

  return <div className={classes.root}>{createContents()}</div>
}
