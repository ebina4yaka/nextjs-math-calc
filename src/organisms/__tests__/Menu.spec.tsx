import React from 'react'
import renderer from 'react-test-renderer'
import Menu from '../Menu'

test('Menu Snapshot Test', () => {
  const tree = renderer.create(<Menu />).toJSON()
  expect(tree).toMatchSnapshot()
})
