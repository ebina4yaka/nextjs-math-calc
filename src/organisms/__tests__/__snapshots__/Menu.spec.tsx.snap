// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Snapshot Test 1`] = `null`;
